ocaml/tests/test_vm_helpers.ml

```diff
@@ -148,40 +148,58 @@ let rec assert_equivalent expected_grouping actual_grouping =
     assert_host_groups_equal e g;
     assert_equivalent es gs
 
-let assert_grouping ~__context gpu_group vgpu_type g =
-  let vgpu_type_ref = Xapi_vgpu_type.find_or_create ~__context vgpu_type in
-  let host_lists = group_hosts_by_best_pgpu_in_group ~__context gpu_group vgpu_type_ref in
-  assert_equivalent g host_lists
-
-let rec assert_expectations ~__context gpu_group = function
+let assert_host_groups_equal_for_vgpu g g' =
+  match g' with
   | [] -> ()
-  | (vgpu_type, expected_grouping) :: remaining ->
-    assert_grouping ~__context gpu_group vgpu_type expected_grouping;
-    assert_expectations ~__context gpu_group remaining
+  | _ ->
+    let extract_host_strings g =
+      let hosts = g in
+      List.sort String.compare (List.map Ref.string_of hosts)
+    in
+    Alcotest.(check (slist string String.compare))
+      "check host strings"
+      (extract_host_strings g) (extract_host_strings g')
 
-let check_expectations __context gpu_group ~k100_lst ~k140q_lst =
-  assert_expectations ~__context gpu_group
-    VGPU_T.[ k100,  k100_lst ; k140q, k140q_lst ]
+let rec assert_equivalent_for_vgpu expected_grouping actual_grouping =
+  match (expected_grouping, actual_grouping) with
+  | [], [] -> ()
+  | [], xx -> Alcotest.fail (Printf.sprintf "%d more groups than expected." (List.length xx))
+  | xx, [] -> Alcotest.fail (Printf.sprintf "%d less groups than expected." (List.length xx))
```

Comment:
Minor grammar issue: it's better to use `fewer` than `less` (because groups are countable).